copy/copy.go

```diff
@@ -121,6 +144,23 @@ type Options struct {
 	Progress         chan types.ProgressProperties // Reported to when ProgressInterval has arrived for a single artifact+offset.
 	// manifest MIME type of image set by user. "" is default and means use the autodetection to the the manifest MIME type
 	ForceManifestMIMEType string
+	// CheckAuthorization is an indication if the image does not need to be decrypted, but only
+	// requires checking if the it can be decrypted.
+	// This is an optimization for cached images to see if provided keys have the authorization
```

Comment:
Cached in this case means that the image exists in the store. This is more relevant in the context of the container runtime, cri-o. In this case, the scenario is where an image had been downloaded and unpacked during `ImagePull`, but we should be given the option to check if the image is decryptable at runtime as a security check. This is to prevent user B from using user A's encrypted image, just because they are on the same host and the image has already been unpacked.